File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-5.cpp
class Solution {
public:
    int findMin(vector<int> &nums) {
        int l = 0;
        int h = nums.size() - 1;
        while (l < h) {
            int m = (h + l)/2;
            if (nums[m] < nums[h]) {
                h = m;
            } else {
                l = m + 1;
            }
        }
        if (nums[l] > nums[0]) {
            return nums[0];
        }
        return nums[l];
    }
};

File: Data Structures & Algorithms/two-integer-sum/submission-5.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> differences = new HashMap<>();
        int[] res = {0, 0};

        for (int i = 0; i < nums.length; i++) {
            if (differences.get(nums[i]) != null) {
                res[0] = differences.get(nums[i]);
                res[1] = i;
                return res;
            }

            differences.put(target - nums[i], i);
        }

        return res;

    }
}
